src/template.c

```diff
@@ -7302,6 +7244,72 @@ bool TemplateInstance::hasNestedArgs(Objects *args, bool isstatic)
     return nested != 0;
 }
 
+/*****************************************
+ * Append 'this' to the enclosing module members[] so the semantic routines
+ * will get called on the instance members.
+ */
+Dsymbols *TemplateInstance::appendToModuleMember()
+{
+    Module *mi = minst;
+    if (mi)
+    {
+        if (mi->isRoot())
+        {
+            /* If both modules mi and md where the template is declared are
+             * roots, instead use md to store the generated code in it.
+             */
+            Module *md = tempdecl->getModule();
+            if (md && md->isRoot())
+                mi = md;
```

Comment:
That seems to be a new rule, why is it needed?
Who actually introduced that whacky minst concept, there can be multiple modules instantiating a template in D?